File: Assets/MapCreation.js
#pragma strict

	public var vs : int = 20;
	public var hs : int = 40;
	public var matrix : int[,]; //0 significa zona habitable
								//dif de 0 zona no habitable
	public var basic_floor : GameObject;
	public var nhabitable_floor : GameObject;

function Start () {
	var basic_floor_instance : GameObject;
	for(var i : int = 0; i < vs; i++){
		for(var j : int = 0; j < hs; j++){
		    if(matrix[i,j]==0) basic_floor_instance = Instantiate(basic_floor, Vector3(i,-0.5,j), Quaternion.identity);
		    else basic_floor_instance = Instantiate(nhabitable_floor, Vector3(i,0.5,j), Quaternion.identity);
	    }
    }
}

function Update () {
	
}

function setMatrix(new_floor : int[,], x : int, z : int){
	matrix = new_floor;
	vs = x;
	hs = z;
}

File: Assets/MapCreation.cs
using UnityEngine;
using System.Collections;

public class MapCreation : MonoBehaviour {
	
	private int vs = 20;
	private int hs = 40;
	public int[,] matrix; 	//0 significa zona habitable
							//dif de 0 zona no habitable
	public GameObject basic_floor;
	public GameObject nhabitable_floor;

	void Start () {
		//Creacion de suelo plano
		GameObject basic_floor_instance = Instantiate(basic_floor, new Vector3(0,-0.5f,0), Quaternion.identity) as GameObject;
		basic_floor_instance.transform.localScale = new Vector3 (vs, 1, hs);
		basic_floor_instance.transform.Translate((vs / 2) - 0.5f, 0, (hs / 2) - 0.5f);
		//Anadir pilares para zonas inaccesibles
		for(int i = 0; i < vs; i++){
			for(int j = 0; j < hs; j++){
				if(matrix[i,j]!=0) basic_floor_instance = Instantiate(nhabitable_floor, new Vector3(i,1f,j), Quaternion.identity) as GameObject;
			}
		}
	}

	public void setMatrix(int[,] new_floor, int x, int z){
		matrix = new_floor;
		vs = x;
		hs = z;
	}
}